### PR TITLE
FIX: Email substution bug

### PR DESCRIPTION
**Pull request by riking**
https://meta.discourse.org/t/regexp-escapes-interpreted-in-emails/184673

This changes from providing a string literal for the #sub replacement, to providing a block.
Because the block is provided the match object, it is presumed to have already performed all necessary backreferences.
This avoids any replacement of backreferences in the message body.

### CODE REVIEW

**Comment by riking** [comment]
Update: Culprit is somewhere in `Email::MessageBuilder.new(...).html_part`

**Comment by riking** [comment]
Debugging this was a miserable swamp of "single" stepping through ActionMailer and discovering that the email had already been fully rendered.

File: app/helpers/email_helper.rb [hunk 1]
```diff
@@ -26,10 +26,11 @@ def email_topic_link(topic)
...
   def email_html_template
-    EmailStyle.new.html
-      .sub('%{email_content}', capture { yield })
-      .gsub('%{html_lang}', html_lang)
-      .html_safe
+    tmp = EmailStyle.new.html
+    tmp = tmp.sub('%{email_content}') { capture { yield } }
+    tmp = tmp.gsub('%{html_lang}', html_lang)
+    tmp = tmp.html_safe
+    tmp
```
Comment (CvX) [inline app/helpers/email_helper.rb:33]: Great find!

Would it be better to just change the `{ capture { yield } }` part? I assume `tmp` stuff was just for debugging purposes?

**Review by CvX — APPROVED** [review]

File: app/helpers/email_helper.rb (hunk 1 above)
Comment (riking) [reply app/helpers/email_helper.rb]: Done

**Review by CvX — APPROVED** [review]

**Pull request merged**